### DISCUSSION TITLE
Can't set the PackageVersion via target

### DISCUSSION THREAD

**Discussion by SongOfYouth** [unoplatform/uno#14380] [Q&A]
### Current behavior

When i set the package version of  uno project vith `<PackageVersion>0.1.0</PackageVersion>` via a target which before `Build` target, it doesn't work, but other properties work such as `FileVersion` and `AssemblyVersion`.

### Workaround

Add this target to your csproject:
```xml
	<Target Name="SetPackageVersion" BeforeTargets="Build">
		<PropertyGroup>
			<PackageVersion>0.1.0</PackageVersion>
		</PropertyGroup>
	</Target>
```

### Works on UWP/WinUI

No

### NuGet package version(s)

```xml
<PackageReference Include="Uno.WinUI" Version="4.10.26" />
```

### IDE

Visual Studio 2022

### IDE version

17.6.6

### Anything else we need to know?

it only works by adding the `<PackageVersion>0.1.0</PackageVersion>`  property into `PropertyGroup` which in the project root node

**Comment by jeromelaban** [comment]
Thanks for the report. This is a behavior is expected and out of the control from Uno Platform, and comes from the .NET SDK. It is very likely that the `PackageVersion` property is copied by a conditional set of `ItemsGroup` in a targets file, executed as constants before `Build` happens.

You can analyze what's happening in msbuild using the [binlog viewer](https://msbuildlog.com/).
